store/sqlstore/scheme_store.go

```diff
@@ -46,48 +46,49 @@ func (s SqlSchemeStore) createIndexesIfNotExists() {
 	s.CreateIndexIfNotExists("idx_schemes_channel_admin_role", "Schemes", "DefaultChannelAdminRole")
 }
 
-func (s *SqlSchemeStore) Save(scheme *model.Scheme) (*model.Scheme, *model.AppError) {
+func (s *SqlSchemeStore) Save(scheme *model.Scheme) (*model.Scheme, error) {
 	if len(scheme.Id) == 0 {
 		transaction, err := s.GetMaster().Begin()
 		if err != nil {
-			return nil, model.NewAppError("SqlSchemeStore.SaveScheme", "store.sql_scheme.save.open_transaction.app_error", nil, err.Error(), http.StatusInternalServerError)
+			return nil, errors.Wrap(err, "begin_transaction")
 		}
 		defer finalizeTransaction(transaction)
 
-		newScheme, appErr := s.createScheme(scheme, transaction)
-		if appErr != nil {
-			return nil, appErr
+		newScheme, err := s.createScheme(scheme, transaction)
+		if err != nil {
+			return nil, err
 		}
 		if err := transaction.Commit(); err != nil {
-			return nil, model.NewAppError("SqlSchemeStore.SchemeSave", "store.sql_scheme.save_scheme.commit_transaction.app_error", nil, err.Error(), http.StatusInternalServerError)
+			return nil, errors.Wrap(err, "commit_transaction")
 		}
 		return newScheme, nil
 	}
 
 	if !scheme.IsValid() {
-		return nil, model.NewAppError("SqlSchemeStore.Save", "store.sql_scheme.save.invalid_scheme.app_error", nil, "schemeId="+scheme.Id, http.StatusBadRequest)
+		// TODO: Here another type of error would be better, waiting for suggestions in review.
```

Comment:
Same here, I think is the right error, but maybe @agnivade have a different opinion.

Reply:
Yes, this is fine. The entire scheme is invalid.